### PR TITLE
Add SimulationClock

### PR DESCRIPTION
**Pull request by schroedtert**
First implementation of a SimulationClock.

### CODE REVIEW

File: cpp/libcore/source/util/simulation_clock.hpp [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
-/// SimulationClock provides methods for python iterator bindings.
-/// In Python we want to iterate over the timesteps of the simulation like:
-/// ```for cur_step in simulation.steps():```
-/// see
-/// https://stackoverflow.com/questions/49257947/iterating-over-c-objects-wrapped-in-python-via-pybind11
-/// and https://github.com/pybind/pybind11/blob/master/tests/test_sequences_and_iterators.cpp
 class SimulationClock
 {
+    const std::uint64_t m_max_steps;
+    const double m_delta_t;
```
Comment (schroedtert) [inline cpp/libcore/source/util/simulation_clock.hpp:18]: Should these be `public`?

File: cpp/pycore/test/util/simulation_clock_test.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import pytest
```
Comment (schroedtert) [inline cpp/pycore/test/util/simulation_clock_test.py:1]: Not sure if here also the functionality should be tested. Seems to be the same as in the cpp tests.

File: cpp/libcore/source/util/simulation_clock.hpp (hunk 1 above)
Comment (Ozaq) [inline cpp/libcore/source/util/simulation_clock.hpp:18]: Is there a reason you are not using `std::chrono::milliseconds` or the sort of?

File: cpp/libcore/source/util/simulation_clock.hpp [hunk 3]
```diff
@@ -1,19 +1,48 @@
...
+    auto getDeltaT() const -> double { return m_delta_t; }
+    auto getCurrentStep() const -> std::uint64_t { return m_current_step; }
+    auto getCurrentTime() const -> double
+    {
+        return static_cast<double>(m_current_step) * m_delta_t;
+    }
+
+    /// Advance the clock to the next step. Increases \p m_current_step
+    /// \throws JPSexception, when \p m_current_step exceeds \p m_max_steps
+    auto advance() -> void
```
Comment (Ozaq) [inline cpp/libcore/source/util/simulation_clock.hpp:40]: If you would change the interface to: `auto advance() -> bool` and any advance that would advance the clock past the max steps return false, you could write the simulation loops as:
```python
while clock.advance():
    simulation.compute_next_step()
```

File: cpp/libcore/source/util/simulation_clock.hpp (hunk 1 above)
Comment (schroedtert) [reply cpp/libcore/source/util/simulation_clock.hpp]: Changed it to `std::chrono::milliseconds`

File: cpp/libcore/source/util/simulation_clock.hpp (hunk 3 above)
Comment (schroedtert) [reply cpp/libcore/source/util/simulation_clock.hpp]: Changed it.

**Review by zeroset — APPROVED** [review]

**Pull request merged**